jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamExecutable.kt

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.sam
+
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.util.text.SemVer
+import software.aws.toolkits.jetbrains.core.executables.AutoResolvable
+import software.aws.toolkits.jetbrains.core.executables.ExecutableCommon
+import software.aws.toolkits.jetbrains.core.executables.ExecutableType
+import software.aws.toolkits.jetbrains.core.executables.Validatable
+import software.aws.toolkits.jetbrains.settings.ExecutableDetector
+import java.nio.file.Path
+import java.nio.file.Paths
+
+class SamExecutable : ExecutableType<SemVer>, AutoResolvable, Validatable {
+    override val displayName: String = "sam"
+    override val id: String = "samCli"
+
+    val samMinVersion = SemVer("0.38.0", 0, 38, 0)
+    val samMaxVersion = SemVer("0.50.0", 0, 50, 0)
```

Comment:
Consider comment or adding into the valName that it is exclusive. Makes for quick reference when looking at the code later on.

Reply:
readded the comments